src/planwise/client/projects/views.cljs

```diff
@@ -92,29 +92,30 @@
 
                 ;; Base tile layer
                 mapping/gray-base-tile-layer
-                ;; Markers with filtered facilities
-                (when (#{:facilities :transport} selected-tab)
-                  [:point-layer {:points @facilities
-                                 :popup-fn marker-popup-fn
-                                 :radius 5
-                                 :color styles/light-grey
-                                 :stroke false
-                                 :fillOpacity 1}])
-                ;; Demographics tile layer
-                (let [demand-map     (when (= :transport selected-tab) (mapping/demand-map @demand-map-key))
-                      population-map (mapping/region-map project-region-id)]
-                  [:wms-tile-layer {:url config/mapserver-url
-                                    :transparent true
-                                    :layers mapping/layer-name
-                                    :DATAFILE (or demand-map population-map)
-                                    :opacity 0.3}])
                 ;; Boundaries of working region
                 (if @map-geojson
                   [:geojson-layer {:data @map-geojson
                                    :color styles/green
                                    :fit-bounds true
                                    :fillOpacity 0.1
                                    :weight 0}])
+                ;; Demographics tile layer
+                (let [demand-map     (when (= :transport selected-tab) (mapping/demand-map @demand-map-key))
+                      population-map (mapping/region-map project-region-id)]
+                  [:wms-tile-layer {:url config/mapserver-url
+                                    :transparent true
+                                    :layers mapping/layer-name
+                                    :DATAFILE (or demand-map population-map)
+                                    :opacity 0.5}])
+                ;; Markers with filtered facilities
+                (when (#{:facilities :transport} selected-tab)
+                  (for [[type facilities] @facilities-by-type]
+                    [:point-layer {:points facilities
+                                   :popup-fn marker-popup-fn
+                                   :radius 4
+                                   :color (:colour type)
+                                   :stroke false
+                                   :fillOpacity 1}]))
```

Comment:
I think it's time to refactor this. I'd implement a separate function to yield the map's layers and splat them `into` a base map, also created separately. This will yield the ugly `mapcat` in `leaflet-update-layers` unnecessary.

The result would look something like this:

```clojure
        [:div
         [sidebar-section selected-tab]
         [:div.map-container
          (let [map-props {:position @map-position
                           :zoom @map-zoom
                           :min-zoom 5
                           :on-position-changed
                           #(dispatch [:projects/update-position %])
                           :on-zoom-changed
                           #(dispatch [:projects/update-zoom %])}
                layers [;; Base tile layer
                        mapping/gray-base-tile-layer
                        ;; Demographics tile layer
                        (let [demand-map ...
                              population-map ...]
                          [:wms-tile-layer ...])
                        ;; Boundaries of working region
                        (if @map-geojson
                          [:geojson-layer ...])
                        ;; Markers with filtered facilities
                        (when (#{:facilities :transport} selected-tab)
                          [:point-layer {:points @facilities
                                         :popup-fn marker-popup-fn
                                         :radius 5
                                         :color styles/light-grey
                                         :stroke false
                                         :fillOpacity 1}])
                        ;; Isochrone for selected transport
                        (when (= :transport selected-tab)
                          [:geojson-bbox-layer ...])]]
            (into [map-widget map-props] (filter some? layers)))]]
```

Obviously this is from before this PR, but the idea is the same. Pre-process the children before yielding the Reagent component.